### ISSUE TITLE
missing defaults for extensible templates location

### ISSUE THREAD

**Issue by lkingland** [knative-sandbox/kn-plugin-func#347]
Either I forgot to put in the logic, or it was inadvertently refactored away:  the `--templates` flag should have logic which first searches for templates using the logic indicated by its comment:
```
  // Templates is an optional path that, if it exists, will be used as a source
  // for additional templates not included in the binary.  If not provided
  // explicitly as a flag (--templates) or env (FUNC_TEMPLATES), the default
  // location is $XDG_CONFIG_HOME/templates ($HOME/.config/func/templates)
  Templates string
```
Note that the funtionality of actually _respecting_ this setting is already in place and tested, we just need the defaulting.

**Comment by lance** [comment]
@lkingland this can be closed now that #415 has landed, correct?

**Comment by lkingland** [comment]
Yes this can, thanks Lance
